core/src/main/java/com/datastax/oss/driver/api/core/metadata/Node.java

```diff
@@ -57,12 +60,16 @@
    * <p>This may not be know at all times. In particular, current Cassandra versions (3.10) only
    * store it in {@code system.local}, so this will be known only for the control node.
    */
+  @NonNull
   Optional<InetSocketAddress> getListenAddress();
 
+  @NonNull
   String getDatacenter();
 
+  @NonNull
   String getRack();
 
+  @NonNull
   Version getCassandraVersion();
```

Comment:
Datacenter, rack and version should be non-null in a healthy deployment, but the driver would still work if the data was corrupted. I'm going to mark them as nullable and add explanatory javadocs.